test/unit/data.js

```diff
@@ -874,3 +874,18 @@ test( "Check that the expando is removed when there's no more data", function()
 		}
 	}
 });
+
+test( ".data(prop) does not create expando", function() {
+	expect( 2 );
+
+	var div = jQuery( "<div/>" );
+
+	div.data("foo");
+	equal( false, jQuery.hasData( div[0] ) );
+	// Make sure no expando has been added
+	for ( key in div[ 0 ] ) {
```

Comment:
> We can codify this, sure, if JSCS can do it. 

It totally can, it was like that for a while, but code-style changed and rule was removed from the preset. It is like that de-facto because code-style force it, but now it doesn't so we should ask contributor to do something that is not documented